### PR TITLE
Adding full support for interfaces

### PR DESCRIPTION
**Pull request by shinmog**
```
  - name: "Verify add"
    panos_interface:
      ip_address: '{{ ip_address }}'
      username: '{{ username }}'
      password: '{{ password }}'
      if_name: 'ethernet1/5'
      zone_name: "ansible"
      create_default_route: 'yes'
      commit: False

  - name: "Verify l3 update to static ip addressing"
    panos_interface:
      ip_address: '{{ ip_address }}'
      username: '{{ username }}'
      password: '{{ password }}'
      operation: 'update'
      if_name: 'ethernet1/5'
      mode: 'layer3'
      ip: ['10.5.5.1/24', '10.6.6.1/24']
      enable_dhcp: False
      create_default_route: False
      zone_name: "ansible"
      commit: False

  - name: "Verify l2 update"
    panos_interface:
      ip_address: '{{ ip_address }}'
      username: '{{ username }}'
      password: '{{ password }}'
      operation: 'update'
      if_name: 'ethernet1/5'
      mode: 'layer2'
      enable_dhcp: False
      zone_name: "ansible-l2"
      commit: False

  - name: "Verify delete"
    panos_interface:
      ip_address: '{{ ip_address }}'
      username: '{{ username }}'
      password: '{{ password }}'
      operation: 'delete'
      if_name: 'ethernet1/5'
      zone_name: "l2-zone"
      commit: False
```

### CODE REVIEW

File: examples/fw_interface.yml [hunk 1]
```diff
@@ -3,22 +3,47 @@
   connection: local
   gather_facts: False
 
-  roles:
-    - role: PaloAltoNetworks.paloaltonetworks
-
-  vars:
-    zone_name: "tap_zone"
+#  roles:
```
Comment (ivanbojer) [inline examples/fw_interface.yml]: Uncomment these. We still need roles until RedHat is synced up with us.

File: library/panos_interface.py [hunk 2]
```diff
@@ -80,98 +168,255 @@
...
     zone_name = module.params['zone_name']
-    create_default_route = module.params['create_default_route']
+    vr_name = module.params['vr_name']
+    vsys_dg = module.params['vsys_dg']
     commit = module.params['commit']
 
-    ifexists = if_exists(xapi, if_name)
+    # Open the connection to the PANOS device.
+    con = base.PanDevice.create_from_device(*auth)
+    # con = base.PanDevice.create_from_device( module.params['ip_address'], module.params['username'], module.params['password'], module.params['api_key'])
```
Comment (ivanbojer) [inline library/panos_interface.py]: Feel free to remove this if *auth works.

**Comment by shinmog** [comment]
I think I've addressed all the feedback at this point, with the exception of unittests..?

**Comment by ivanbojer** [comment]
Done!

**Pull request merged**